viewing_party/party.py

```diff
@@ -0,0 +1,199 @@
+    #and a value which is a list of dictionaries representing the movies the user wants to watch
+    #user_data={'watchlist':[{}{}{}]}
+    watchlist=user_data['watchlist']
+    watchlist.append(movie)
+    return user_data
+  
+def watch_movie(user_data,title):
+       #user_data={'watchlist':[{}{}{}], 'watched':[{}{}{}]}
+    list=[]
+    to_watch_movies=user_data['watchlist']
+    for i in range(len(to_watch_movies)):
+      to_watch_movie=to_watch_movies[i]['title']
+      list.append(to_watch_movie)
+
+    if title in list:
+        #i have to remove and add the movie corresponding to this title
+        movie_index=list.index(title)
+        x=to_watch_movies.pop(movie_index)
+        already_watched_movie=user_data['watched']
+        already_watched_movie.append(x)
+        return user_data
+    else: 
+        return user_data
+
+## WAVE 2 !!!
+
+
+def get_watched_avg_rating(user_data):
+    #user_data={'watched'=[{},{},{}]}
+    list_of_ratings=[]
+    watchedlist=user_data['watched']
+    for i in range(len(watchedlist)):
+      rating=user_data['watched'][i]['rating']
+      list_of_ratings.append(rating)
+    if len(watchedlist)==0:
+      avg_rating=0.0
+    else:
+      avg_rating=sum(list_of_ratings)/len(watchedlist)
+    
+    return avg_rating
+
+
+def get_most_watched_genre(user_data):
+    #user_data={'watched'=[{'genre'},{}]}
+    list_of_genres=[]
+    watchedlist=user_data['watched']
+    for i in range(len(watchedlist)):
+      genre=user_data['watched'][i]['genre']
+      list_of_genres.append(genre)
+    thisset=set(list_of_genres)
+    count_list=[]
+    genre_list=[]
+    for i in range(len(thisset)):
+      each_genre=list_of_genres[i]
+      Count=list_of_genres.count(each_genre)
+      genre_list.append(each_genre)
+      count_list.append(Count) #i want this to reflect the actual genre not the number
+    if len(watchedlist)==0:
+      return None
+    else:
+      most_frequent=max(count_list)
+      index_of_most_frequent=count_list.index(most_frequent)
+      max_genre=genre_list[index_of_most_frequent]
+      return max_genre
+
+def get_unique_watched(user_data):
+  #user_data={'watched':[{},{}],'friends':[{'watched':[{'title':,'genre':,'rating':},{}]},{}]
+
+  user_watched=user_data['watched']
+  #which will return a list of dictionaries of my films
+  #i will then loop through to get all titles
+
+  user_watched_titles=[]
+  for i in range(len(user_watched)):
+      user_watched_title=user_watched[i]
+      user_watched_titles.append(user_watched_title)
+
+  friends_watched_titles=[]
+  friends_watched=user_data['friends']
+  for j in range(len(friends_watched)):
+    friend_ind_watched=friends_watched[j]['watched']
+    for k in range(len(friend_ind_watched)):
+      friends_watched_titles.append(friend_ind_watched[k])
+  unique_watched=[]
+  for title in user_watched_titles:
+    if title not in friends_watched_titles:
+      unique_watched.append(title)
+  
+  return unique_watched
```

Comment:
this works well but I wanted to show a way to do this using `movie["title"]`
```suggestion
def get_unique_watched(user_data):
    friends_watched_titles = []
    unique_watched = []

    for watched_list in user_data["friends"]:
        for movie in watched_list["watched"]:
            friends_watched_titles.append(movie["title"])

    for movie in user_data["watched"]:
        if movie["title"] not in friends_watched:
            unique_watched.append(movie_item)

    return unique_watched 
```